php-frontend/src/main/java/org/sonar/php/tree/impl/declaration/ParameterTreeImpl.java

```diff
@@ -147,7 +155,8 @@ public Kind getKind() {
   public Iterator<Tree> childrenIterator() {
     return IteratorUtils.concat(
       attributeGroups.iterator(),
-      IteratorUtils.iteratorOf(visibility, readonlyToken, type, referenceToken, ellipsisToken, variableIdentifier, equalToken, initValue));
+      visibilityAndReadonly.iterator(),
```

Comment:
Do I understand correctly that the problem was that we always submitted `visibility` and then `readonly` tokens, even though they could have been in an opposite order? And basically the fix is to keep them as a collection and thus keep the original order?

Reply:
Exactly. Unfortunately, the grammar and order are not clear.